sycl/test/on-device/back_to_back_collectives.cpp

```diff
@@ -0,0 +1,64 @@
+// RUN: %clangxx -fsycl -fsycl-unnamed-lambda -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %RUN_ON_HOST %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+
+#include <CL/sycl.hpp>
+#include <numeric>
+#include <vector>
+using namespace cl::sycl;
+using namespace cl::sycl::ONEAPI;
+
+int main() {
+  queue q;
+  if (q.get_device().is_host()) {
+    std::cout << "Skipping test\n";
+    return 0;
+  }
+
+  // Use max work-group size to maximize chance of race
+  int N = q.get_device().get_info<info::device::max_work_group_size>();
```

Comment:
Probably it's not a best choice. This test is failing on CPU device with "out of resources" error.

Reply:
You're right.  I forgot that calling the work-group functions might impact the maximum work-group size supported by the kernel.  I've made the query kernel-specific in https://github.com/intel/llvm/pull/2990/commits/6d9918b07eed9b58749014d6af16f8b41c86392e.